### PR TITLE
Increasing modularity and algorithm separated logic

### PR DESCRIPTION
**Pull request by d4nilos**
In this PR, we started to break the AutoscaleJob class and increase the project modularity.
We are trying to open the possibility to add new algorithms for autoscaling logic.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,9 +49,9 @@ Start the service with docker-compose using a dockerized local postgres:
 Register the Bigtable cluster that should be autoscaled in the service:
 
 ```
-PROJECT_ID=<YOUR GCP PROJECT ID>
-INSTANCE_ID=<YOUR INSTANCE ID>
-CLUSTER_ID=<YOUR CLUSTER ID>
+PROJECT_ID=autoscaler-hackweek
```
Comment (bobo) [inline README.md]: fix

File: src/main/java/com/spotify/autoscaler/AutoscaleJob.java [hunk 2]
```diff
@@ -361,25 +259,39 @@ void run(
...
       }
     }
     final Duration samplingDuration = getSamplingDuration(cluster, timeSupplier);
-    int newNodeCount =
-        cpuStrategy(cluster, clusterResizeLogBuilder, samplingDuration, currentNodes);
-    newNodeCount =
-        storageConstraints(
-            cluster, clusterResizeLogBuilder, samplingDuration, newNodeCount, currentNodes);
+
+    ScalingEvent newScalingEvent =
```
Comment (bobo) [inline src/main/java/com/spotify/autoscaler/AutoscaleJob.java]: This looks kinda awfull, but not sure how to improve it :-(

File: src/main/java/com/spotify/autoscaler/db/BigtableCluster.java [hunk 3]
```diff
@@ -72,3 +74,8 @@ default boolean exists() {
     return errorCode != ErrorCode.GRPC_NOT_FOUND && errorCode != ErrorCode.PROJECT_NOT_FOUND;
   }
 }
+
+enum LastChangeType {
```
Comment (bobo) [inline src/main/java/com/spotify/autoscaler/db/BigtableCluster.java]: remove, it was me thinking...

File: src/main/java/com/spotify/autoscaler/db/BigtableCluster.java [hunk 4]
```diff
@@ -41,6 +41,8 @@
 
   Optional<Instant> lastChange();
 
+  Optional<LastChangeType> lastChangeType();
```
Comment (bobo) [inline src/main/java/com/spotify/autoscaler/db/BigtableCluster.java]: also remove

File: src/main/java/com/spotify/autoscaler/AutoscaleJob.java (hunk 2 above)
Comment (d4nilos) [reply src/main/java/com/spotify/autoscaler/AutoscaleJob.java]: :(

**Pull request merged**